runtime/major_gc.c

```diff
@@ -742,6 +742,8 @@ static intnat mark(intnat budget) {
   return budget;
 }
 
+static enum scanning_action_flags darken_scanning_flags = 0;
```

Comment:
IIUC the state here is `!SCANNING_ONLY_YOUNG_VALUES`. For readability, would it not be better to have another constant in the `scanning_action_flags` enum?

Reply:
I prefer 0. I think of these flags as a set of atomic flags of value 2^k -- not an enum 1,2,3..., but really flags 1,2,4,8... By construction, 0 is the empty set; using it feels appropriate in connection with the test code (`flags & FOO`) which exposes the assumption that intersection is bitwise-and, and non-emptyness is non-0ness. Does that make sense?

Reply:
Thanks for the clarification.

Reply:
A bitfield is a good idea, but usually you still define 0 as "None" (or something) in the enum for this purpose.